### PR TITLE
[FEATURE] enable setting config via configuration

### PR DESCRIPTION
**Pull request by geraldaistleitner**
For CD it should be possible to specify the LDAP configuration via configuration files. This feature is now implemented and can be seen in the Documentation -> Developers

### CODE REVIEW

File: Classes/Domain/Repository/ConfigurationRepository.php [hunk 1]
```diff
@@ -49,6 +49,11 @@ public function findAll()
 
         $rows = static::getDatabaseConnection()->exec_SELECTgetRows('*', $this->table, $where, '', 'sorting');
 
+        $config = unserialize($GLOBALS['TYPO3_CONF_VARS']['EXT']['extConf']['ig_ldap_sso_auth']);
+        if ($config['useExtConfConfiguration']) {
```
Comment (xperseguers) [inline Classes/Domain/Repository/ConfigurationRepository.php]: We should use `if (!empty($config...)) {`

File: Classes/Domain/Repository/ConfigurationRepository.php [hunk 2]
```diff
@@ -68,9 +73,14 @@ public function findAll()
      */
     public function findByUid($uid)
     {
-        $where = 'uid=' . (int)$uid . $this->getWhereClauseForEnabledFields();
+        $config = unserialize($GLOBALS['TYPO3_CONF_VARS']['EXT']['extConf']['ig_ldap_sso_auth']);
```
Comment (xperseguers) [inline Classes/Domain/Repository/ConfigurationRepository.php]: Here as well. But why unserializing each time? Better would be to have it once for all in the constructor.

File: Documentation/Development/Index.rst [hunk 3]
```diff
@@ -23,3 +23,56 @@ Hooks
 	Hooks/ExtraDataProcessing
 	Hooks/ExtraMergeField
 	Hooks/GetGroupsProcessing
+
+Continuous Deployment
+-----
```
Comment (xperseguers) [inline Documentation/Development/Index.rst]: Underline should match the length of the title and have an empty line afterwards

File: Documentation/Development/Index.rst [hunk 4]
```diff
@@ -23,3 +23,56 @@ Hooks
...
+To support continuous deployment, it is possible to set a configuration via PHP through the following code in Additional_Configuration.php or ext_localconf.php of your extension.
+
+.. toctree::
```
Comment (xperseguers) [inline Documentation/Development/Index.rst]: why `.. toctree::` ?

File: Resources/Private/Language/locallang_db.xlf [hunk 5]
```diff
@@ -12,6 +12,9 @@
 			<trans-unit id="settings.forceLowerCaseUsername">
 				<source>Force lower-case user name: If checked, user names will be always be stored in lower-cases. Useful for case-sensitive databases.</source>
 			</trans-unit>
+			<trans-unit id="settings.useExtConfConfiguration">
```
Comment (xperseguers) [inline Resources/Private/Language/locallang_db.xlf]: XLIFF key should not use abbreviations like "ext" or "conf"

File: Documentation/Development/Index.rst (hunk 4 above)
Comment (xperseguers) [reply Documentation/Development/Index.rst]: This is `AdditionalConfiguration.php`. Please use proper reST markup for referencing files. In addition, you use the ConfigurationManager to set the configuration of the extension, but that's not how one is supposed to write custom configuration in `AdditionalConfiguration.php`. Why not directly assigning the value to the global array as it is usually done?

**Comment by geraldaistleitner** [comment]
Squashed the commits.

**Comment by geraldaistleitner** [comment]
Hello Xavier, 
is there still something wrong with my PR?
Best regards, Gerald

**Pull request merged**